composer/callbacks/early_stopper.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Early stopping callback."""
+
+from __future__ import annotations
+
+import logging
+from typing import Callable
+
+import numpy as np
+
+from composer.core import State
+from composer.core.callback import Callback
+from composer.loggers import Logger
+
+log = logging.getLogger(__name__)
+
+__all__ = ["EarlyStopper"]
+
+
+class EarlyStopper(Callback):
+    __doc__ = f"""Callback to halt training early.
+    """
+
+    def __init__(
+        self,
+        monitor: str,
+        eval_label: str,
+        comp: Callable = None,
+        min_delta: float = 0.0,
+        patience: int = 1,
+    ):
+        self.monitor = monitor
+        self.eval_label = eval_label
+        self.comp = comp
+        self.min_delta = abs(min_delta)
+        if self.comp is None:
+            self.comp = np.less if 'loss' in monitor.lower() or 'error' in monitor.lower() else np.greater
+            if self.comp == np.less:
+                self.min_delta *= -1
+
+        if self.comp(1, 2):
```

Comment:
a custom `comp` might not actually work with integers, so this could be dangerous. I think we can skip setting these and let the user take control here.